### PR TITLE
Use FT for Trusted OS verification

### PR DESCRIPTION
**Pull request by AlCutter**
This PR changes the bootloader to use FT to verify the Trusted OS instead of relying on signatures over the ELF.

This is a departure from the original plan of sticking with multiple vanilla signatures on the ELF, but the extra complexity is low as we're sharing FT verification code from `-common` and we can roll this back before first production devices if we're not comfortable with it, and I've added "ifdef'd out" code which shows how we'd do signature verification only over the manifest.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -66,47 +78,106 @@ func preLaunch() {
...
+	fw = &firmware.Bundle{
+		Checkpoint:     conf.Bundle.Checkpoint,
+		Index:          conf.Bundle.LogIndex,
+		InclusionProof: conf.Bundle.InclusionProof,
+		Manifest:       conf.Bundle.Manifest,
+	}
+
+	fw.Firmware, err = card.Read(conf.Offset, conf.Size)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read firmware: %v", err)
```
Comment (roger2hk) [inline main.go:108]: Error strings should not be capitalized.

File: main.go [hunk 2]
```diff
@@ -116,6 +187,27 @@ func main() {
...
+		panic(fmt.Sprintf("armored-witness-boot: load error, %v\n", err))
+	}
+}
+
+func manifestVerifiers() ([]note.Verifier, error) {
+	var manifestKeys []string
+	if err := json.Unmarshal([]byte(OSManifestVerifiers), &manifestKeys); err != nil {
+		return nil, fmt.Errorf("invalid OSManifestVerifiers format: %v", err)
 	}
+	manifestVerifiers := []note.Verifier{}
```
Comment (roger2hk) [inline main.go:199]: Would the size be always equal to the `len(manifestKeys)`?

File: main.go [hunk 3]
```diff
@@ -66,47 +78,106 @@ func preLaunch() {
 	usbarmory.LED("white", false)
 }
 
+// read reads the firmware bundle for the trusted OS from internal storage, the
+// OS proof bundle is *not* verified by this function.
+func read(card *usdhc.USDHC) (fw *firmware.Bundle, err error) {
+	blockSize := card.Info().BlockSize
+	if blockSize != expectedBlockSize {
+		return nil, fmt.Errorf("h/w invariant error - expected MMC blocksize %d, found %d", expectedBlockSize, blockSize)
```
Comment (roger2hk) [inline main.go:86]: Where you would write “actual” and “expected”, prefer using the words “got” and “want”, respectively.
https://google.github.io/styleguide/go/decisions.html#got-before-want

**Review by roger2hk — APPROVED** [review]

File: main.go (hunk 2 above)
Comment (AlCutter) [reply main.go]: Yup, changed to preallocate the right size.

**Review by roger2hk — APPROVED** [review]

**Pull request merged**